test/e2e/specs/treeview/00-static-treeview.spec.js

```diff
@@ -37,23 +37,132 @@ exports.tests = function (appName, baseUrl) {
           expect(searchDiv.isDisplayed()).toBeTruthy("Search box input is not visible when page loads initially.");
         });
 
-        it('expand all and collapse all button are visible', function(){
+      });
+
+      describe('actions for the page', function() {
+        beforeAll(function() {
+          browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout);
+          filterDropDown.click();
+          browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout)
+          element(by.id('ui-id-2')).click();
+
+          deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
+
+        });
+
+        it('all node list collapse on `Collapse All` click', function(done) {
+          actionBtn.all(by.css('button')).then(function(buttons) {
+            buttons[1].click().then(function() {
+              expect(element(by.id('jstree')).all(by.css('.jstree-open')).count()).toBe(0);
+              done();
+            });
+          }).catch(function(err) {
+            console.log(err);
+          });
+        });
+
+
+        it('all node list expand on `Expand All` click', function(done) {
           actionBtn.all(by.css('button')).then(function(buttons) {
-            expect(buttons.length).toEqual(2, '2 buttons are not visible.');
-            expect((buttons[0]).getText()).toEqual('Expand All', 'Expand All button is not correct.');
-            expect((buttons[1]).getText()).toEqual('Collapse All', 'Collapse All button is not correct.');;
+            buttons[0].click().then(function() {
```

Comment:
This test case fails for me most of the time with this message:
`Expected 0 to be 4.`

I'm guessing it's checking for the "sub-tree" elements before they have finished loading in the DOM. Try adding a wait for condition here. You could try 2 different things, the 1st should work, the 2nd definitely will:
1. Wait for `.jstree-open` node to be visible
    ```js
    browser.wait(EC.presenceOf(element(by.id('jstree')).first(by.css('.jstree-open'))))
    ```

2. Wait for all `.jstree-open` nodes to be visible
    ```js
     browser.wait(function() {
       return element(by.id('jstree')).all(by.css('.jstree-open').count().then(function(ct) {
         return (ct==4);
       });
    }, browser.params.defaultTimeout);
    ```

The same comment about promise chaining applies here too.